### PR TITLE
AO3-4836 500 error when approving moderated comments on work you didn't create

### PR DESCRIPTION
**Pull request by zz9pzza**
## Issue

https://otwarchive.atlassian.net/browse/AO3-4836

## Purpose

Testing the comments controller though up an issue and this should clean it up. Merge after #2734 (which was previously #2715)

## Testing

Refer to Jira, although this is also covered by an automated test.

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -163,10 +163,13 @@ def index
 
   def unreviewed
     if @commentable.nil?
+      # I think you can't get here as the earlier tests in check_permission_to_review
```
Comment (sarken) [inline app/controllers/comments_controller.rb]: I agree -- this code can go

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -50,74 +48,54 @@ def find_parent
...
-  
+
   # Comments cannot be edited after they've been replied to
   def check_permission_to_edit
-    unless @comment && @comment.count_all_comments == 0
-      flash[:error] = ts('Comments with replies cannot be edited')
-      redirect_to(request.env["HTTP_REFERER"] || root_path) and return
-    end
+    return if @comment && @comment.count_all_comments == 0
+    flash[:error] = ts('Comments with replies cannot be edited')
```
Comment (sarken) [inline app/controllers/comments_controller.rb]: As long as we're cleaning all the things, can you make this double quotes?

**Comment by sarken** [comment]
Merge conflicts /o\

**Comment by zz9pzza** [comment]
Failing Scenarios:
cucumber features/importing/work_import_da.feature:32 # Scenario: Creating a new art work from a deviantart gallery link fails - it needs the direct link
cucumber features/importing/work_import_da.feature:42 # Scenario: Creating a new fic from deviantart import

Are standard right now.

**Pull request merged**